Feature/47 results/table

MichaelXian opened the pull request:
Still uses hardcoded data until the backend api is implemented (unless that already happened and I just didn't notice)

cormacjmollitor committed:
Added the Bootstrap CSS and JS files in the client folder

Going to work on the search results page now

cormacjmollitor committed:
Used CDN for Bootstrap and jQuery Instead

Will delete the bootstrap and jquery files in a later commit.

MichaelXian committed:
Create ResultsTable class

MichaelXian committed:
Add ugly table

Created a functional table with dummy data, however, it currently looks
ugly. Will work on making it look better in the future.

MichaelXian committed:
Make the table more beautiful and add feautres

Made the table look as similar to the wireframe as I could. Items can
now be selected. I also added the original posting/apply buttons. There
is an excerpt that shows before being selected, and the full description
is shown after being selected. From what I can tell from the wireframe,
the location is supposed to disappear when selected. This has been
implemented. Also added fontAwesome stylesheet, but I can't figure out
how to import it, so it's just kinda there for when I figure out later,
so I can make the open table arrow look better.

MichaelXian commented:
I meant search bar, not backend api

frostyshadows commented on client/src/App.js:
```diff
@@ -25,15 +33,11 @@ class App extends Component {
-          </p>
-        </header>
-        <p>{this.state.data}</p>
+        <Navbar></Navbar>
```
can write `<Navbar/>` instead

MichaelXian replied:
<code>\<Navbar /\></code> has been written

frostyshadows commented on client/src/index.js:
```diff
@@ -2,5 +2,6 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './css/index.css';
 import App from './App';
+//import './css/fontAwesome.css'; How do
```
what's this for?

MichaelXian replied:
I wanted to use an icon for the opening of tables, but i can't figure out how to import the css file

MichaelXian replied:
Removed failed attempt at importing icons

frostyshadows commented on client/src/js/Navbar.js:
```diff
@@ -0,0 +1,79 @@
+    data: null
+  };
+/*
+  componentDidMount() {
```
You can delete this commented out part for now, I don't think we'll need an API call for the Navbar

MichaelXian replied:
Deletions have been made

frostyshadows commented on client/src/js/ResultsTable.js:
```diff
@@ -0,0 +1,82 @@
+
+  handleClick(action, result) {
+    switch (action) {
+      case 0: // Clicked on row to select/deselect
```
Javascript switch statements can switch on strings so you could use a descriptive string for the action instead of just 0, 1, and 2. Using numbers can get confusing.

MichaelXian replied:
Replaced 0, 1, 2 with "select", "post", and "apply"

frostyshadows reviewed (changes requested):
The Navbar and Results table look great when I run it! Left a few suggested changes. I also got a few errors printed to the console, if you haven't looked into them yet please see if they can be fixed. 
<img width="1316" alt="screen shot 2018-10-25 at 9 05 49 pm" src="https://user-images.githubusercontent.com/12488887/47543992-c9adb300-d899-11e8-8dc4-917a46aa906e.png">

DConnorFong reviewed (changes requested):
I see added files for bootstrap, were we not still using CDNs for everything temporarily?

MichaelXian committed:
Resolve some requests

Resolved all errors in the console except failing to load the backend,
as I assume this will succeed in loading the backend in the future. The
commented failed import has been removed, as has the commented API call
in Navbar. The switch/case in ResultsTable has been changed to use
strings instead. <Navbar></Navbar> has been replaced with <Navbar />.

MichaelXian committed:
Remove bootstrap and fontAwesome.css
